src/common/components/markdownViewer/markdownViewerComponent.tsx

```diff
@@ -1,33 +1,110 @@
 import * as React from 'react';
-import { marksy } from 'marksy';
+import * as ReactDOM from 'react-dom';
+import { withRouter } from 'react-router';
+import { SOURCE_LINE_CLASSNAME,
+         getPixelOffsetForSourceLine,
+         getSourceLineForPixelOffset } from './syncScroll';
+import { CreateMarkdownRender, Mdr, MdrFactory,
+         MdrSetup, MdrOptions, MdrCodeStyle } from './render';
+import { throttle } from '../../../common/helper/limitExecution';
 
-const compile = marksy({
-  // TODO: extract into a new <DynamicLink /> that analyze "href" and render a <Link /> or <a />
-  a: ({ target, children, ...other }) => (
-    <a target="_self" {...other}>{children}</a>
-  ),
-});
+/**
+ * TODO:
+ * Custom rule to analyze "href" and render a <Link /> (react router) or <a />
+ */
 
-export interface MarkDownViewerComponentProps {
+interface MarkDownViewerComponentProps {
   content: string;
+  scrollSourceLine?: number;
+  onScrollSourceLine?: (sourceLine) => any;
+  className?: string;
+  location?: any; // Router HOC injected.
+  options?: MdrOptions;
+  codeStyle?: MdrCodeStyle;
 }
 
-const getMarkDownChildren = (content: string): React.ReactNode => {
-  let childrenComponent: React.ReactNode = null;
+class MarkDownViewer extends React.Component<MarkDownViewerComponentProps, {}> {
+  constructor(props) {
+    super(props);
+  }
 
-  if (content) {
-    childrenComponent = compile(content).tree;
+  private CreateMdrInstance = () => {
+    const setupParams: MdrSetup = {
+      routerLocation: this.props.location ? this.props.location.pathname : '',
```

Comment:
why is it necessary routerLocation?

Reply:
Do you remember that I told you I managed to fix scroll navigation to #id elements? That is something not possible with react-router by default. Well, markdonw-it-footnote adds links in the markdown automatically based on #id. So, in order for this feature to work properly, I must know the current router location in order to automatically add it in the anchor link.

It is better to explain it graphically. Check `Footnotes` section in the markdown sample text. 
![image](https://user-images.githubusercontent.com/24454225/30758777-0bb090e2-9fd4-11e7-9dfb-b56b00b887de.png)

By default, the auto-generated links would be rendered in HTML as:
```html
<a href="#fn1" id="fnref1">[1]</a>
```
But we need router location to be added as prefix to work properly:
```html
<a href="#/trainer/training/1/edit#fn1" id="fnref1">[1]</a>
```

Reply:
ok

Reply:
Ok great. However, for some reason, it stopped working today. It's been a couple of months since I committed this code and maybe something has changed in the meantime. 

[Personal reminder] TODO: Fix it.